web-server/src/orm/entities/users/User.ts

```diff
@@ -1,9 +1,16 @@
-import bcrypt from 'bcryptjs';
-import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
-
-import { Role, Language } from './types';
+import { 
+  Entity,
+  Check,
+  PrimaryGeneratedColumn,
+  Column,
+  CreateDateColumn,
+  OneToMany,
+} from 'typeorm';
+import { Country } from './types';
+import { Task } from '../tasks/Task'
 
 @Entity('users')
+@Check('"username" ~* "[a-zA-Z]+_-\..*[0-9]+"')  // Unverified regex recipe
```

Comment:
It might be cleaner to use this library:
https://github.com/typestack/class-validator

Arguably, this is a TypeORM API problem because they force all the validators to be outside the class where it makes no sense.